pyani_plus/utils.py

```diff
@@ -145,13 +145,39 @@ def _fmt_cmd(args: list[str]) -> str:
 
 
 def check_output(args: list[str]) -> str:
-    """Wrap for subprocess.check_output and report any error to stderr."""
+    """Wrap for subprocess.run and report any error to stderr.
+
+    Note that if the output is not of interest on success, subprocess.check_call
+    would be natural instead. However, the documentation for that recommends not
+    to use this with subprocess.PIPE in case the child process blocks.
+    """
+    code = 0
+    output = None
     try:
         return subprocess.check_output(
             [str(_) for _ in args], stderr=subprocess.STDOUT, text=True
         )
     except subprocess.CalledProcessError as err:
-        sys.stderr.write(f"ERROR: Return code {err.returncode} from {_fmt_cmd(args)}\n")
-        if err.output:
-            sys.stderr.write(err.output)
-        sys.exit(err.returncode)
+        code = err.returncode
+        output = err.output
+
+    msg = f"ERROR: Return code {code} from: {_fmt_cmd(args)}\n"
+    sys.stderr.write(msg)
+    if output:
+        sys.stderr.write("~" * 60 + "\n")
+        sys.stderr.write(output)
+        sys.stderr.write("~" * 60 + "\n")
+        sys.stderr.write(msg)  # repeat as this is important
```

Comment:
Should the comment `# repeat as this is important` be on line 170 or 169?

As usual, I have a question 😄. Why is it important to repeat this? There's no need to provide a long explanation in the code, but if you think it's necessary, feel free. I'm just curious, as always.

Reply:
I've repeated the key bit at top and bottom because I've seen the output cropped for display (e.g. the default exceptions via typer). And generally when long text gets cropped you either see the start or the end. But yes, not entirely logical 😉 

This touches on an area where legacy pyANI was extremely frustrating to use - I'm hoping we can do better at capturing and displaying the errors from individual jobs in pyani-plus.

Reply:
Thanks @peterjc for explaining this. It makes more sens now.